### PR TITLE
JAudio2 Work

### PR DESCRIPTION
**Pull request by jdflyer**
also some Z2WolfHowlMgr work

### CODE REVIEW

File: include/JSystem/JAudio2/JAISound.h [hunk 1]
```diff
@@ -29,33 +29,106 @@ struct JASTrack {
     /* 80291C30 */ void openChild(u32);
     /* 80292918 */ void writePort(u32, u16);
     /* 8029297C */ void readPort(u32);
+
+    inline int getChannelMgrCount() {
```
Comment (TakaRikka) [inline include/JSystem/JAudio2/JAISound.h:33]: the inline keyword isnt needed if the definition is in the class

File: include/JSystem/JAudio2/JAISound.h [hunk 2]
```diff
@@ -29,33 +29,106 @@ struct JASTrack {
...
+    }
+    inline void fadeOut(u32 fadeCount) {
+        if (fadeCount!=0) {
+            mTransition.set(0.0f,mIntensity,fadeCount);
+        }else{
+            forceOut();
+        }
+    }
+    inline bool isOut() {
+        if(mTransition.mCount!=0||mIntensity<0.01f) {
```
Comment (TakaRikka) [inline include/JSystem/JAudio2/JAISound.h:121]: while i dont wanna stress formatting issues too much yet at this point in the project, it still would be good if spacing consistency can be kept like `if (cond1 != 0 || cond2 < 0.01f)` and `func(param0, param1, param2)`

File: include/JSystem/JAudio2/JASGadget.h [hunk 3]
```diff
@@ -6,9 +6,35 @@
...
+                sInstance = (T*)this;
+                //We need a better way to compute the location of sInstance
+                //sInstance = (T*)((char*)this-(char*)&(((T*)NULL)->JASGlobalInstance<T>));
+            //}
+        }
+    }
+    static T* sInstance;
 };
 
+class Parent {
```
Comment (TakaRikka) [inline include/JSystem/JAudio2/JASGadget.h:24]: is this temp stuff that can be removed?

Comment (jdflyer) [reply include/JSystem/JAudio2/JASGadget.h:24]: Yeah, let me get that out of there

**Review by Pheenoh — APPROVED** [review]

**Pull request merged**